components/server/ee/src/workspace/workspace-factory.ts

```diff
@@ -117,6 +117,7 @@ export class WorkspaceFactoryEE extends WorkspaceFactory {
                 ws = await this.createForCommit({span}, user, commitContext, normalizedContextURL);
             }
             ws.type = "prebuild";
+            ws.projectId = project?.id;
```

Comment:
I would still argue that this belongs into `createForPrebuiltWorkspace` (as optional parameter) but that's not too important here, and this fixes it as well.